packages/typescript/src/components/EnumDeclaration.tsx

```diff
@@ -11,11 +12,31 @@ import { useTSNamePolicy } from "../name-policy.js";
 import { createTSSymbol, useTSScope } from "../symbols/index.js";
 import { BaseDeclarationProps } from "./Declaration.js";
 import { EnumMember } from "./EnumMember.jsx";
+import { JSDoc } from "./JSDoc.jsx";
+
+/**
+ * A descriptor for an enum member.
+ */
+export interface EnumMemberDescriptor {
+  /**
+   * The JavaScript value of the enum member.
+   */
+  jsValue: string | number;
+  /**
+   * Documentation for the enum member.
+   */
+  doc?: string | string[];
+}
+
 export interface EnumDeclarationProps extends BaseDeclarationProps {
   /**
-   * A JS object representing the enum member names and values.
+   * The members of the enum.
+   */
+  members?: Record<string, string | number | EnumMemberDescriptor>;
+  /**
+   * Documentation for the enum.
    */
-  jsValue?: Record<string, string | number>;
```

Comment:
@bterlson This would be a breaking change to allow values or descriptor. What do you think?